### DISCUSSION TITLE
Batpred + Podpoint Solo EV Charger - getting this to work!

### DISCUSSION THREAD

**Discussion by LAScotland** [springfall2008/batpred#649] [Q&A]
Hey folks,

I've got a running batpred system, with a Podpoint solo charger, and a Vauxhall corsa E.

so far so good - everything is up and running, i'm using these integrations for car / charger:

https://github.com/mattrayner/pod-point-home-assistant-component - fully functional
https://github.com/flobz/psacc-ha/tree/main/psacc-ha- currently borked

I've got the cars section of my app.yaml like so:

`  car_charging_energy: 're:(sensor.myenergi_zappi_[0-9a-z]+_charge_added_session|sensor.wallbox_portal_added_energy|sensor.psl_529952_total_energy)'

  num_cars: 1

  car_charging_planned:
    - 're:(sensor.wallbox_portal_status_description|sensor.myenergi_zappi_[0-9a-z]+_plug_status|sensor.psl_529952_status)'

  car_charging_planned_response:
    - 'yes'
    - 'on'
    - 'true'
    - 'connected'
    - 'ev connected'
    - 'charging'
    - 'paused'
    - 'waiting for car demand'
    - 'waiting for ev'
    - 'scheduled'
    - 'enabled'
    - 'latched'
    - 'locked'
    - 'plugged in'

  #car_charging_now:
  #  - off

  car_charging_now_response:
    - 'yes'
    - 'on'
    - 'true'

  #car_charging_battery_size:
  #  - 75
  #car_charging_limit:
  #  - 're:number.tsunami_charge_limit'
  #car_charging_soc:
  #  - 're:sensor.tsunami_battery'

  octopus_intelligent_slot: 're:(binary_sensor.octopus_energy([0-9a-z_]+|)_intelligent_dispatching)'
  octopus_ready_time: 're:(time.octopus_energy([0-9a-z_]+|)_intelligent_ready_time)'
  octopus_charge_limit: 're:(number.octopus_energy([0-9a-z_]+|)_intelligent_charge_limit)'
`

Which is removing the car charge data from the power plan, great - I'd really like to take this to the next level though.

As i understand the Pod Point  integration, what I need to do is write a new schedule to the charger each day, probably with the most economic contigious 4 hour charge slot (in lieu of the PSA integration working again, if it ever does, and reporting a SoC for the battery) and then writing it to the next day. The simple "on/off" switch doesn't seem to work (or at least has some interaction with the scheduling part).

Is there something I can pull out of Predbat which will help with this, or has anyone managed to get predbat to take over the scheduling entirely?

**Comment by springfall2008** [comment]
Have you seen my video on this topic?

https://youtu.be/805Blbfc9Os

**Reply by LAScotland** [reply]
Hi @springfall2008 I did - thankyou very much, it was very helpful - confirmed that what I had done already was correct. Thanks for predbat in general - it's great!

I had already got around kicking off a charge (weirdly podpoint requires that it is in scheduled mode, effectively with no schedule for the charge override to function)- my complication is that I don't have a "end charge override" service on the pod point integration.

All I can do is kick off a charge, for a specified period of time, so for now I'm just going for 30 minutes, and non combined slots - it's not super reliable though, as you say in your video.

How would you go around creating a helper to get the binary_sensor.predbat_car_charging_slot.planned[0].duration in order to allow a dynamic charge length?

![image](https://github.com/springfall2008/batpred/assets/95568985/ff8d7a9b-b3c8-4fe0-82eb-812439fd6166)

**Reply by Nordtech** [reply]
Hi Luke, came across your question because I too have a Pod Point Solo and am gingerly feeling my way around using Predbat and setting it up for a 2nd car. Did you get any further with this?
I have a Tesla and a VW e-Up. The Tesla is currently managed by Octopus IG with the Pod Point app being in Smart mode and all that appears fine in Predbat. For the VW, I put the PP app in Manual and set a schedule (11:30-05:30) so now I'm editing the apps.yaml to add it in. One thing I noticed was that Pod Point EV Charger - Car Charge Mode entity had the values
    - 'available'
    - 'unavailable'
    - 'charging'
    - 'out-of-service'
    - 'waiting-for-schedule'
    - 'connected-waiting-for-schedule'
   so Ive set car_charging_planned_response to these - not sure if thats right no not....
Cheers
